### PR TITLE
chore: trim unused parts of rendezvous info

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

I've always been annoyed that I never understood what all of the fields of RendezvousInfo were for.  I grep'd the code and realized only a couple were ever being used.

## Test Plan

I plan on testing the full suite of distributed tests before landing this.

### CODE REVIEW

File: harness/tests/experiment/utils.py [hunk 1]
```diff
@@ -154,9 +154,7 @@ def make_default_env_context(
 
 
 def make_default_rendezvous_info() -> det.RendezvousInfo:
-    return det.RendezvousInfo(
-        addrs=["127.0.0.1:1750"], addrs2=[f"127.0.0.1:{constants.LOCAL_RENDEZVOUS_PORT}"], rank=0
-    )
+    return det.RendezvousInfo(addrs=["127.0.0.1:1734"], rank=0)
```
Comment (shiyuann) [inline harness/tests/experiment/utils.py:157]: Should use the constant `LOCAL_RENDEZVOUS_PORT` here.

**Review by shiyuann — APPROVED** [review]

**Review by stoksc — COMMENTED** [review]
lgtm, but to understand the implications of the change I have a historcal question I can't remember the answer to: why did we ever have 2 rendezvous ports? I feel like aaron explained to me at one point but I honestly don't remember.

**Comment by rb-determined-ai** [comment]
> why did we ever have 2 rendezvous ports?

There was a time that we had way more network connections in the harness doing more things.  I don't remember what we needed the second one for exactly but we don't need it anymore.

File: master/internal/trial.go [hunk 2]
```diff
@@ -131,15 +131,8 @@ func (m *trialMessage) UnmarshalJSON(data []byte) error {
 }
 
 type rendezvousInfoMessage struct {
-	// Addrs is deprecated in favor of Containers.
```
Comment (stoksc) [inline master/internal/trial.go:134]: lol

Comment (stoksc) [reply master/internal/trial.go:134]: that went well

**Comment by rb-determined-ai** [comment]
GKE and GPU and all parallel tests all passed in the last push: https://app.circleci.com/pipelines/github/determined-ai/determined/12908/workflows/c26cbec5-345c-4698-b0aa-ad2b476e02e5

And this push only changed a line in the unit tests, and unit tests all passed.

Landing this now.

**Pull request merged**